src/classes/OPP_OpportunityContactRoles_TDTM.cls

```diff
@@ -375,22 +375,29 @@ public without sharing class OPP_OpportunityContactRoles_TDTM extends TDTM_Runna
         // Retrieve the Account Ids for the Organizational Opportunities
         Map<Id, List<Id>> accountsToOpportunities = populateAccountToOpportunities(listOpps, oppAccountDetails);
 
-        //find changed opportunities and instantiate a map entry to hold OCRs
-        for (integer i=0; i<listOpps.size(); i++) {
-            if (listOpps[i].Primary_Contact__c != listOldOpps[i].Primary_Contact__c ||
-                ((listOpps[i].AccountId != listOldOpps[i].AccountId)
-                    && (!oppAccountDetails.get(listOpps[i].AccountId).npe01__SYSTEMIsIndividual__c || !oppAccountDetails.containsKey(listOldOpps[i].Id))) ||
-                needsManageOCR(listOpps[i], listOldOpps[i], FIELD_NAME_HONOREE_CONTACT) ||
-                needsManageOCR(listOpps[i], listOldOpps[i], FIELD_NAME_NOTIFICATION_RECIPIENT_CONTACT)) {
+        // find changed opportunities and instantiate a map entry to hold OCRs
+        for (Integer i=0; i<listOpps.size(); i++) {
+            Opportunity newOpp = listOpps[i];
+            Opportunity oldOpp = listOldOpps[i];
+
+            Boolean isIndividualAccount = (newOpp.AccountId != null && oppAccountDetails.containsKey(newOpp.AccountId))
```

Comment:
What do you think about pulling this into it's own method? We have the same variable declared 3x, the [first one](https://github.com/SalesforceFoundation/Cumulus/blob/4f58258775c723310459ad808f10488eb9631726/src/classes/OPP_OpportunityContactRoles_TDTM.cls#L251) doesn't have the accountid null check but not sure if it's really needed either?

Reply:
Good suggestion.

Reply:
I also agree with @jjbennett suggestion for putting the code for determining the value of the isIndividualAccount in its own method now that the code is being called twice in a method. However, I understand minimizing changes. 

I'd also recommend adding the null check to the isIndividualAccount code in the `upsertContactRolesAfterInsert` method.